Makefile

```diff
@@ -114,23 +115,57 @@ lib: prepare_lib
 		STACK_DEV_OPTIONS="--fast --ghc-options -Wwarn"
 
 metadata : x_token_symbol = x
-metadata : x_token_name = "Token X"
+metadata : x_token_name = Token X
 metadata : x_token_decimals = 1
 metadata : y_token_symbol = y
-metadata : y_token_name = "Token Y"
+metadata : y_token_name = Token Y
 metadata : y_token_decimals = 1
 metadata : output = metadata.json
-metadata: lib
+metadata :
+	# ============== Generate TZIP16 metadata ============== #
 	$(MAKE) -C haskell exec PACKAGE=segmented-cfmm \
 		EXEC_ARGUMENTS="print-metadata \
 		--x-token-symbol $(x_token_symbol) \
-		--x-token-name $(call escape_double_quote,$(x_token_name)) \
+		--x-token-name \"$(call escape_double_quote,$(x_token_name))\" \
 		--x-token-decimals $(x_token_decimals) \
-		--y-token-symbol $(y_token_symbol)
-		--y-token-name $(call escape_double_quote,$(y_token_name)) \
+		--y-token-symbol $(y_token_symbol) \
+		--y-token-name \"$(call escape_double_quote,$(y_token_name))\" \
 		--y-token-decimals $(y_token_decimals) \
 		" EXEC_OUTPUT=$(output)
 
+metadata_byte : x_token_symbol = x
+metadata_byte : x_token_name = Token X
+metadata_byte : x_token_decimals = 1
+metadata_byte : y_token_symbol = y
+metadata_byte : y_token_name = Token Y
+metadata_byte : y_token_decimals = 1
+metadata_byte : output = metadata_byte.json
+metadata_byte :
+	# ============== Generate TZIP16 metadata in hex ============== #
+	$(MAKE) -C haskell exec PACKAGE=segmented-cfmm \
+		EXEC_ARGUMENTS="print-metadata-byte \
+		--x-token-symbol $(x_token_symbol) \
+		--x-token-name \"$(call escape_double_quote,$(x_token_name))\" \
+		--x-token-decimals $(x_token_decimals) \
+		--y-token-symbol $(y_token_symbol) \
+		--y-token-name \"$(call escape_double_quote,$(y_token_name))\" \
+		--y-token-decimals $(y_token_decimals)" EXEC_OUTPUT=$(OUT)/metadata_byte.json
+
+metadata_uri : metadata_key = sub_metadata
+metadata_uri :
+	# ============== Generate metadata_uri to be used in tezos-storage ============== #
+	$(MAKE) -C haskell exec PACKAGE=segmented-cfmm \
+		EXEC_ARGUMENTS="generate-metadata-uri \
+		--metadata-uri-key $(metadata_key)" EXEC_OUTPUT=$(OUT)/metadata_uri.json
+
+metadata_bigmap: output_file = $(OUT)/metadata_map
+metadata_bigmap: metadata_byte metadata_uri
+	echo "((Big_map.literal [ \
+		(\\\"\\\", Bytes.pack \\\"$(shell cat $(OUT)/metadata_uri.json)\\\") ; \
+		(\\\"$(metadata_key)\\\", Bytes.pack \\\"$(shell cat $(OUT)/metadata_byte.json)\\\") \
+		]) : metadata_map)" > $(output_file)
```

Comment:
same as above
```suggestion
$(OUT)/metadata_map: $(OUT)/metadata_byte $(OUT)/metadata_uri
	echo "((Big_map.literal [ \
		(\\\"\\\", Bytes.pack \\\"$(shell cat $(OUT)/metadata_uri)\\\") ; \
		(\\\"$(metadata_key)\\\", Bytes.pack \\\"$(shell cat $(OUT)/metadata_byte)\\\") \
		]) : metadata_map)" > $@
```

Reply:
`Bytes.pack` should not be used here AFAIK.

Isn't the reason why we do the conversion to hex ourselves that we want to put byte literals here?

Reply:
Just wanna add a slightly related point: to make sure we're encoding the metadata correctly, we could use the [`checkView` test helper](https://gitlab.com/morley-framework/morley-metadata/-/blob/30f690ba78659fc0648b4c788b92b724eb9354d3/code/morley-metadata-test/src/Test/Morley/Metadata.hs#L27-35) from the `morley-metadata-test` package.

Reply:
There are some [usage examples in stablecoin](https://github.com/tqtezos/stablecoin/search?q=checkView), if that helps.

Reply:
Make sense, I've removed the `Bytes.pack` and use hex directly now. 

Not sure how I'll add the test yet, I'll try it out.

Reply:
@dcastro I've added the test to check if the metadata is encoded properly now, you can check it again.